package/MDAnalysis/lib/pkdtree.py

```diff
@@ -127,22 +130,22 @@ def set_coords(self, coords, cutoff=None):
 
         """
         # If no cutoff distance is provided but PBC aware
-        if self.pbc and (cutoff is None):
-            raise RuntimeError('Provide a cutoff distance'
-                               ' with tree.set_coords(...)')
 
         # set coords dtype to float32
         # augment coordinates will work only with float32
         coords = np.asarray(coords, dtype=np.float32)
 
         if self.pbc:
-            self.cutoff = cutoff
+            if self.cutoff is None:
+                raise ValueError("Cutoff needs to be provided "
+                                 "when working with PBC.")
```

Comment:
In this specific instance, because you move an existing exception raising, it would be good to raise the same exception as what you removed:

```suggestion
                raise RuntimeError('Provide a cutoff distance'
                               ' with tree.set_coords(...)'
```

Reply:
Done